contributor-book/src/guides/onnx-to-burn-conversion-tool.md

```diff
@@ -73,6 +75,193 @@ To extend `burn-import` with support for new ONNX operators, follow these steps:
    Further details can be found in the
    [onnx-tests README](https://github.com/tracel-ai/burn/blob/6d96e8d8086d2309c425f2c8a43a8246f8c454d2/crates/burn-import/onnx-tests/README.md).
 
+## Implementing a New Operator
+
+To extend the capabilities of the Burn library by supporting new operations imported from ONNX
+graphs, developers must go through a few systematic steps. Here, we detail the process, using the
+implementation of the `Squeeze` operation to illustrate points as needed. All file/directory paths
+are relative to `burn/crates/burn-import/`.
+
+### Step 1: Visibility
+
+To make a new operation accessible to the rest of the Burn project, you need to declare the module
+within the
+[`mod.rs` file](https://github.com/tracel-ai/burn/blob/9c5b07c833865bff7f82431001076a33d0d8729c/crates/burn-import/src/burn/node/mod.rs#L24)
+located in the `src/burn/node/` directory.
+
+### Step 2: Node Implementation
+
+Create a new file named `<operation_name>.rs` in the `src/burn/node/` directory.  
+This file will define the structure and functionality of your new operation. By convention, the
+necessary information for carrying out an operation is encapsulated within a struct named
+`<operation>Node`. For the `Squeeze` operation, we defined a
+[struct called `SqueezeNode`](https://github.com/tracel-ai/burn/blob/9c5b07c833865bff7f82431001076a33d0d8729c/crates/burn-import/src/burn/node/squeeze.rs#L8)
+that holds necessary information about the input tensor, output tensor, and axes for the operation.
+
+The core of integrating a new operation involves implementing the `NodeCodegen` trait for your node.
+This trait defines how the node generates code during the graph compilation process. The
+implementation must provide methods to define input and output types, to generate the forward pass
+code, and to encapsulate the node into the more general `Node` structure. Specifically:
+
+- `output_types` and `input_types` return the tensor (or element) types for the output and inputs of
+  the node, respectively.
+- `forward` generates the Rust code that performs the operation during the execution phase. The
+  `quote!` macro is used to generate rust code. Ensure that this is syntactically correct using Burn
+  code.
+- `into_node` wraps the specific node in a general `Node` type, facilitating its inclusion in the
+  broader Burn graph structure.
+
+This file is also where you would put `test_codegen_nodes()`, to make sure that the generated code
+works within the Burn library.
```

Comment:
Awesome!

Can we add a small asterisk for binary and unary ops? The codegen is mostly implemented for these nodes in `binary.rs` and `unary.rs`, and each new op only has to define a method to execute the function on the input(s).

No need to detail the process as much here, just a small note that it is slightly different for those but it works in the same way.

Reply:
@laggui Absolutely, will add!